### DISCUSSION TITLE
Understanding extra (<none>: <none>) layers

### DISCUSSION THREAD

**Discussion by hjw** [BretFisher/ama#195] [Q&A]
I am confused about differences which happen when I build a local Dockerfile and when I do
docker run <...>

Specifically: 
`docker run -it inikolaev/alpine-scheme`
results in what looks like one image being downloaded and a container being run.

```
doing docker image ls -a results in:
REPOSITORY                TAG                 IMAGE ID            CREATED             SIZE
inikolaev/alpine-scheme   latest              0c0fa9d6fee2        3 weeks ago         77.8MB
```

but, if I then download the Dockerfile from inikolaev's github repo and build it with 
`docker build -t hjw-scheme .
`
`docker image ls -a` results in:

```
REPOSITORY                TAG                 IMAGE ID            CREATED             SIZE
<none>                    <none>              a305b92290e2        6 seconds ago       78.5MB
hjw-scheme                latest              e9adeb8b17c3        6 seconds ago       78.5MB
<none>                    <none>              a5b3f8c3b009        55 seconds ago      5.53MB
inikolaev/alpine-scheme   latest              0c0fa9d6fee2        3 weeks ago         77.8MB
alpine                    latest              5cb3aa00f899        3 weeks ago         5.53MB
```

Why/ what's going on here?

**Comment by BretFisher** [comment]
Your `-a` shows you intermediate images, which are images your images need (the FROM of your image), but docker doesn't label them because you didn't pull them specifically.

Docker will also remove labels if you pull/build an image that is an update of an existing image by the same name. It will remove the label from existing and apply to the new image, but docker doesn't delete images by default. That's what `docker image prune` is for :)
